yt_dlp/YoutubeDL.py

```diff
@@ -4077,6 +4088,22 @@ def _opener(self):
         handler = self._request_director.handlers['Urllib']
         return handler._get_instance(cookiejar=self.cookiejar, proxies=self.proxies)
 
+    def _get_available_impersonate_targets(self):
+        # todo(future): make available as public API
+        return [
+            (target, rh.RH_NAME)
+            for rh in self._request_director.handlers.values()
+            if isinstance(rh, ImpersonateRequestHandler)
+            for target in rh.supported_targets
+        ]
+
+    def _impersonate_target_available(self, target):
+        # todo(future): make available as public API
+        return any(
+            rh.is_supported_target(target)
+            for rh in self._request_director.handlers.values()
+            if isinstance(rh, ImpersonateRequestHandler))
```

Comment:
These are only used in 1 place each. Why even have them as functions? If we want to make them public, we can refactor at the time.

Reply:
They are exposed for an extractor API that will come after

Reply:
The public API signature will likely be different. So better to refactor at the time imo. But not a blocking issue